Add ability to run this project in Docker

uzakov opened the pull request:
**Summary**
This PR makes it easier to run this project, by running it in Docker and not having to install all the Python requirements. As you see from Reddit WSB comments many people can't run this project easily.      

**Note**
This doesn't address PIP8 issues

Commit:
Added Dockerfile in order to run this project easier and faster, expanded README

Commit:
Enhanced description on how to run it and what is required

RKosow commented:
I was just considering doing the same thing!
If you want to make this even more idiot-proof, you could clone the repo in the container and also wrap all your commands with a shell script or make. 
Give them two targets, `make scan` and `make graph`.

BeauBouchard commented on README.md:
```diff
@@ -2,24 +2,45 @@
+## How to run
+Build docker image by running
+<pre>
+docker build docker build --tag NAME:VERSION . ie docker build --tag wsb:1.0 .
```
```suggestion
docker build --tag NAME:VERSION . ie docker build --tag wsb:1.0 .
```

adeiva commented:
This isn’t working for me


Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

From: Beau Bouchard<mailto:notifications@github.com>
Sent: Thursday, July 30, 2020 7:37 PM
To: SamPom100/UnusualVolumeDetector<mailto:UnusualVolumeDetector@noreply.github.com>
Cc: Subscribed<mailto:subscribed@noreply.github.com>
Subject: Re: [SamPom100/UnusualVolumeDetector] Add ability to run this project in Docker (#9)


@BeauBouchard commented on this pull request.


In README.md<https://github.com/SamPom100/UnusualVolumeDetector/pull/9#discussion_r463375162>:

>



--run the market_scanner.py and it will print out results into the terminal



--you can also graph any ticker's volume in grapher.py



+3) Open project in your IDE and install all required dependancies by typing pip install -r requirements.txt into the IDE's terminal. You can get to the the terminal on VSC by pressing CMD and ` at the same time.



+



+## How to run



+Build docker image by running



+<pre>



+docker build docker build --tag NAME:VERSION . ie docker build --tag wsb:1.0 .


⬇️ Suggested change

-docker build docker build --tag NAME:VERSION . ie docker build --tag wsb:1.0 .



+docker build --tag NAME:VERSION . ie docker build --tag wsb:1.0 .



—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/SamPom100/UnusualVolumeDetector/pull/9#pullrequestreview-458882965>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AD5FYCU4NTOCT57M57OBQ4LR6IU6HANCNFSM4PPAEFMA>.

BeauBouchard commented on README.md:
```diff
@@ -2,24 +2,45 @@
+3) Open project in your IDE and install all required dependancies by typing pip install -r requirements.txt into the IDE's terminal. You can get to the the terminal on VSC by pressing CMD and ` at the same time.
+
+## How to run
+Build docker image by running
```
you may want to explain how to install docker, or even link to the install page here: 
https://docs.docker.com/get-docker/

RKosow replied:
That's already noted in the requirements section, line #7 of the readme.
Though it's probably worth being explicit that you need to run these docker commands in the same directory as the Dockerfile.

uzakov replied:
@BeauBouchard See line 7 and @RKosow it's very obvious but also a good idea, adding that now

NihilistPenguin committed:
Update market_scanner.py

Change time output to minutes instead of seconds
Add ability to print all results together an the end
Output anomalous tickers to a results.txt file

NihilistPenguin committed:
Add files via upload

opens anomalous tickers in Robinhood in a new browser window

NihilistPenguin committed:
Update grapher.py

Creates graphs for all tickers in results.txt 
Allows option for single ticker or tickers of your choice

NihilistPenguin committed:
Add files via upload

Example results.txt file

NihilistPenguin committed:
Update README.md

change line numbers for controlling the script

NihilistPenguin committed:
Update README.md

Add/update descriptions grapher.py and openRobinhood.py

NihilistPenguin committed:
Update README.md

Fix formatting error

NihilistPenguin committed:
Update grapher.py

Provides option to save graphs to the figures directory or to print them to screen

NihilistPenguin committed:
Update grapher.py

Create figures dir if it doesn't exist

NihilistPenguin committed:
Update market_scanner.py

remove previous addition of printing anomalous results together at the end

Commit:
add figures directory with demo images

Commit:
add figures directory with sample images

NihilistPenguin committed:
Delete .gitignore

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

Commit:
Merge pull request #16 from tcosculluela/master

Parallelization

SamPom100 committed:
Revert "Parallelization"

Commit:
Merge pull request #20 from SamPom100/revert-16-master

Revert "Parallelization"

Commit:
Merge pull request #18 from NihilistPenguin/master

Add various functionality

SamPom100 committed:
Revert "Add various functionality "

Commit:
Merge pull request #21 from SamPom100/revert-18-master

Revert "Add various functionality "

uzakov commented:
@adeiva Can you please provide more information what exactly is not working for you.
Just tested, all works 
![image](https://user-images.githubusercontent.com/11824569/89063051-677fbd00-d35f-11ea-9b58-10b0781c190c.png)

adeiva commented:
The market scanner runs but it just tells me the amount of time it takes to run.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

From: Alikhan Uzakov<mailto:notifications@github.com>
Sent: Friday, July 31, 2020 10:52 AM
To: SamPom100/UnusualVolumeDetector<mailto:UnusualVolumeDetector@noreply.github.com>
Cc: adeiva<mailto:vdeiva@hotmail.com>; Mention<mailto:mention@noreply.github.com>
Subject: Re: [SamPom100/UnusualVolumeDetector] Add ability to run this project in Docker (#9)


@adeiva<https://github.com/adeiva> Can you please provide more information what exactly is not working for you.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/SamPom100/UnusualVolumeDetector/pull/9#issuecomment-667253038>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AD5FYCTU4AF2ON5EE5KCKGLR6MAEPANCNFSM4PPAEFMA>.

uzakov commented:
@adeiva I just checked again and it all works, have you waited when running the script? It does take a while 
![image](https://user-images.githubusercontent.com/11824569/89065850-584f3e00-d364-11ea-80d0-93ff482a4384.png)

Commit:
As per PR comments removed duplicate code and made it clearer about running the code

Commit:
Merge upstream

uzakov commented:
@SamPom100 I believe its ready for review

adeiva commented:
Yes I’ve waited while running the script and it’s now working.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

From: Alikhan Uzakov<mailto:notifications@github.com>
Sent: Friday, July 31, 2020 11:31 AM
To: SamPom100/UnusualVolumeDetector<mailto:UnusualVolumeDetector@noreply.github.com>
Cc: adeiva<mailto:vdeiva@hotmail.com>; Mention<mailto:mention@noreply.github.com>
Subject: Re: [SamPom100/UnusualVolumeDetector] Add ability to run this project in Docker (#9)


@adeiva<https://github.com/adeiva> I just checked again and it all works, have you waited when running the script? It does take a while
[image]<https://user-images.githubusercontent.com/11824569/89065850-584f3e00-d364-11ea-80d0-93ff482a4384.png>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/SamPom100/UnusualVolumeDetector/pull/9#issuecomment-667289547>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AD5FYCRNENIL23H7MCRRKLTR6MEYNANCNFSM4PPAEFMA>.

gaby commented on Dockerfile:
```diff
@@ -0,0 +1,17 @@
+FROM python:3
```
Consider using smaller image, there's no need to have all of python to run this.

python:3-slim-buster is around 48MB compressed.
python:3 is around 320MB compressed.
